Extensions/BBText/bbtextruntimeobject.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Displays a rich text using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).
+ * @memberof gdjs
+ * @class BBTextRuntimeObject
+ * @extends RuntimeObject
+ */
+gdjs.BBTextRuntimeObject = function(runtimeScene, objectData) {
+  gdjs.RuntimeObject.call(this, runtimeScene, objectData);
+
+  /** @type number */
+  this._opacity = objectData.content.opacity;
+  /** @type boolean */
+  this._visible = objectData.content.visible;
+  /** @type string */
+  this._text = objectData.content.text;
+  /** @type string */
+  this._color = objectData.content.color;
+  /** @type string */
+  this._fontFamily = objectData.content.fontFamily;
+  /** @type number */
+  this._fontSize = objectData.content.fontSize;
+  /** @type boolean */
+  this._wordWrap = objectData.content.wordWrap;
+  /** @type number */
+  this._wrappingWidth = 0;
+  /** @type string */
+  this._align = objectData.content.align;
+
+  if (this._renderer)
+    gdjs.BBTextRuntimeObjectRenderer.call(this._renderer, this, runtimeScene);
+  else
+    this._renderer = new gdjs.BBTextRuntimeObjectRenderer(this, runtimeScene);
+
+  // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+  this.onCreated();
+};
+
+gdjs.BBTextRuntimeObject.prototype = Object.create(
+  gdjs.RuntimeObject.prototype
+);
+gdjs.BBTextRuntimeObject.thisIsARuntimeObjectConstructor = 'BBText::BBText';
+
+gdjs.BBTextRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.BBTextRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    //this is absolutely required to set the wrapping width to the instance's width
+    this.setWrappingWidth(initialInstanceData.width);
+  }
+};
+
+gdjs.BBTextRuntimeObject.prototype.onDestroyFromScene = function(runtimeScene) {
+  gdjs.RuntimeObject.prototype.onDestroyFromScene.call(this, runtimeScene);
+};
+
+gdjs.BBTextRuntimeObject.prototype.update = function(runtimeScene) {
+  if (this._renderer) this._renderer.ensureUpToDate();
+};
+
+/**
+ * Set/Get BBText base style properties
+ */
+gdjs.BBTextRuntimeObject.prototype.setBBText = function(text) {
+  this._text = text;
+  this._renderer.updateText();
+};
+
+gdjs.BBTextRuntimeObject.prototype.getBBText = function() {
+  return this._text;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setColor = function(rgbColorString) {
+  const splitValue = rgbColorString.split(';');
+  if (splitValue.length !== 3) return;
+  const hexColor = `#${gdjs.rgbToHex(
+    parseInt(splitValue[0]),
```

Comment:
Always always always use parseInt with the second argument ("radix"), otherwise a prefix like 0x or 0 will be interpreted as hexadecimal or octal. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt